<h1> Lab 1 </h1>
Akshita Gundavarapu 

In [ ]:
import pandas as pd

<h2> Part 1 </h2>
Question 1

In [135]:
time = pd.read_csv("data /time-series-covid19-deaths-us.csv", sep = "\t")
time

UID iso2 iso3  code3     FIPS      Admin2 Province_State  \
0     84001001   US  USA    840   1001.0     Autauga        Alabama   
1     84001003   US  USA    840   1003.0     Baldwin        Alabama   
2     84001005   US  USA    840   1005.0     Barbour        Alabama   
3     84001007   US  USA    840   1007.0        Bibb        Alabama   
4     84001009   US  USA    840   1009.0      Blount        Alabama   
...        ...  ...  ...    ...      ...         ...            ...   
3337  84056039   US  USA    840  56039.0       Teton        Wyoming   
3338  84056041   US  USA    840  56041.0       Uinta        Wyoming   
3339  84090056   US  USA    840  90056.0  Unassigned        Wyoming   
3340  84056043   US  USA    840  56043.0    Washakie        Wyoming   
3341  84056045   US  USA    840  56045.0      Weston        Wyoming   

     Country_Region        Lat       Long_  ... 12/31/21  1/1/22  1/2/22  \
0                US  32.539527  -86.644082  ...      160     160     160   
1                US  30.727750  -87.722071  ...      593     593     593   
2                US  31.868263  -85.387129  ...       81      81      81   
3                US  32.996421  -87.125115  ...       95      95      95   
4                US  33.982109  -86.567906  ...      198     198     198   
...             ...        ...         ...  ...      ...     ...     ...   
3337             US  43.935225 -110.589080  ...       14      14      14   
3338             US  41.287818 -110.547578  ...       31      31      31   
3339             US   0.000000    0.000000  ...        0       0       0   
3340             US  43.904516 -107.680187  ...       37      37      37   
3341             US  43.839612 -104.567488  ...       14      14      14   

      1/3/22  1/4/22  1/5/22  1/6/22  1/7/22  1/8/22  1/9/22  
0        160     161     161     161     162     162     162  
1        593     594     594     598     601     603     603  
2         81      81      81      83      83      83      83  
3         95      95      95      95      95      95      95  
4        198     198     198     198     199     202     202  
...      ...     ...     ...     ...     ...     ...     ...  
3337      14      14      14      14      14      14      14  
3338      31      34      34      34      34      34      34  
3339       0       0       0       0       0       0       0  
3340      37      39      39      39      39      39      39  
3341      14      16      16      16      16      16      16  

[3342 rows x 731 columns]

Question 2

In [5]:
time.shape[0]
#number of rows 

3342

In [87]:
time.shape[1]
#number of coloumns

731

Question 3

In [8]:
time.dtypes

UID         int64
iso2       object
iso3       object
code3       int64
FIPS      float64
           ...   
1/5/22      int64
1/6/22      int64
1/7/22      int64
1/8/22      int64
1/9/22      int64
Length: 731, dtype: object

Question 4

In [11]:
subset = time[["FIPS", "Admin2", "Province_State", "3/1/20", "1/9/22"]]

In [13]:
subset.head(5)

FIPS   Admin2 Province_State  3/1/20  1/9/22
0  1001.0  Autauga        Alabama       0     162
1  1003.0  Baldwin        Alabama       0     603
2  1005.0  Barbour        Alabama       0      83
3  1007.0     Bibb        Alabama       0      95
4  1009.0   Blount        Alabama       0     202

In [14]:
subset.shape[0]
#number of rows

3342

In [137]:
subset.shape[1]
#number of coloumns

5

Question 5

In [42]:
subset.isna().sum()

FIPS              10
Admin2             6
Province_State     0
3/1/20             0
1/9/22             0
dtype: int64

<h2> Part 2 </h2>
Question 1

In [149]:
grouped = subset.groupby("Province_State").sum()
grouped = grouped.drop(["3/1/20", "FIPS"], axis =1) 
most_deaths = grouped.sort_values("1/9/22", ascending = False)
most_deaths

1/9/22
Province_State                  
California                 77056
Texas                      76502
Florida                    62688
New York                   60611
Pennsylvania               37665
Georgia                    31521
Illinois                   30794
Ohio                       30072
Michigan                   29955
New Jersey                 29456
Arizona                    24704
Tennessee                  20941
Massachusetts              20568
North Carolina             19619
Indiana                    19614
Alabama                    16624
Missouri                   16509
Virginia                   15651
Louisiana                  15054
South Carolina             14746
Oklahoma                   12592
Kentucky                   12358
Maryland                   12211
Wisconsin                  11402
Minnesota                  10906
Mississippi                10527
Colorado                   10425
Washington                 10004
Connecticut                 9377
Arkansas                    9314
Nevada                      8509
Iowa                        8019
Kansas                      7097
New Mexico                  5983
Oregon                      5761
West Virginia               5421
Idaho                       4237
Utah                        3887
Nebraska                    3468
Puerto Rico                 3373
Rhode Island                3121
Montana                     2917
South Dakota                2522
Delaware                    2365
North Dakota                2074
New Hampshire               2019
Maine                       1603
Wyoming                     1572
District of Columbia        1227
Hawaii                      1105
Alaska                       980
Vermont                      483
Guam                         274
Virgin Islands                89
Northern Mariana Islands      19
Grand Princess                 3
Diamond Princess               0
American Samoa                 0

Question 2

In [108]:
type(most_deaths)

pandas.core.frame.DataFrame

In [109]:
most_deaths.index

Index(['California', 'Texas', 'Florida', 'New York', 'Pennsylvania', 'Georgia',
       'Illinois', 'Ohio', 'Michigan', 'New Jersey', 'Arizona', 'Tennessee',
       'Massachusetts', 'North Carolina', 'Indiana', 'Alabama', 'Missouri',
       'Virginia', 'Louisiana', 'South Carolina', 'Oklahoma', 'Kentucky',
       'Maryland', 'Wisconsin', 'Minnesota', 'Mississippi', 'Colorado',
       'Washington', 'Connecticut', 'Arkansas', 'Nevada', 'Iowa', 'Kansas',
       'New Mexico', 'Oregon', 'West Virginia', 'Idaho', 'Utah', 'Nebraska',
       'Puerto Rico', 'Rhode Island', 'Montana', 'South Dakota', 'Delaware',
       'North Dakota', 'New Hampshire', 'Maine', 'Wyoming',
       'District of Columbia', 'Hawaii', 'Alaska', 'Vermont', 'Guam',
       'Virgin Islands', 'Northern Mariana Islands', 'Grand Princess',
       'Diamond Princess', 'American Samoa'],
      dtype='object', name='Province_State')

In [110]:
most_deaths.values

array([[77056],
       [76502],
       [62688],
       [60611],
       [37665],
       [31521],
       [30794],
       [30072],
       [29955],
       [29456],
       [24704],
       [20941],
       [20568],
       [19619],
       [19614],
       [16624],
       [16509],
       [15651],
       [15054],
       [14746],
       [12592],
       [12358],
       [12211],
       [11402],
       [10906],
       [10527],
       [10425],
       [10004],
       [ 9377],
       [ 9314],
       [ 8509],
       [ 8019],
       [ 7097],
       [ 5983],
       [ 5761],
       [ 5421],
       [ 4237],
       [ 3887],
       [ 3468],
       [ 3373],
       [ 3121],
       [ 2917],
       [ 2522],
       [ 2365],
       [ 2074],
       [ 2019],
       [ 1603],
       [ 1572],
       [ 1227],
       [ 1105],
       [  980],
       [  483],
       [  274],
       [   89],
       [   19],
       [    3],
       [    0],
       [    0]])

Question 3

In [144]:
most_deaths[most_deaths.index == "Washington"]

1/9/22
Province_State        
Washington       10004

The total deaths of washington state as of the date 1/9/22 is 10004

Question 4

In [142]:
grouped1 = subset.groupby("Province_State").mean()
grouped1 = grouped1.drop(["3/1/20", "FIPS"], axis =1) 
avg_deaths = grouped1.sort_values("1/9/22", ascending = False)
avg_deaths
avg_deaths[avg_deaths.index == "Washington"]

1/9/22
Province_State        
Washington       244.0

The average deaths of Washington state as of the date 1/9/22 is 240

In [114]:
subset["1/9/22"].mean()

250.6265709156194

The average deaths of the nation as of 1/9/22 is 250

<h2> Part 3 </h2>
Question 1

We need more data to make any further conclusions because there may be other factirs like age, population, vaccination rate or other foctors that may be effecting the average death rate of the state. Looking at more such data can gives us a better picture of why the average death rate of Washington state is slightly lower than the average death rate of the nation. 


Question 2

The attributes of FIPS and Admin2 had 10 and 6 NA values respectively as found from questions in part 1. From the documentation from this data, we understand that FIPS means Federal Information Processing Standards code that uniquely identifies counties within the USA. It is applicable for the US only and Admin2 is County name again US only. This could explain why some of the values in our subset were NA because they might be cases from states or provinces outside of the united sates. 